src/zenml/steps/utils.py

```diff
@@ -452,6 +454,33 @@ def check_output_types_match(
                 f"{getattr(self, PARAM_STEP_NAME)}"
             )
 
+    def _collect_requirements(
```

Comment:
Can we refactor this in some function which can be used both here and in the step executor operator (which has code very similar to this)?

Reply:
@schustmi  I think it's possible but somehow I would want to leave it like this for the moment as refactor this, would change some stuff related to the step that would require some more testing :)